sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -344,6 +344,17 @@ class SQLAppStatusListener(
     update(exec)
   }
 
+  private def onAdaptiveAccumUpdates(event: SparkListenerSQLAdaptiveAccumUpdates): Unit = {
+    val SparkListenerSQLAdaptiveAccumUpdates(executionId, accumIdsToMetricType) = event
+
+    val stages = liveExecutions.get(executionId).stages
+    stages.foreach { stageId =>
+      val liveStageMetric = stageMetrics.get(stageId)
+      stageMetrics.put(stageId, liveStageMetric.copy(
```

Comment:
This can be caused throughout the entire subquery execution, and by doing this "copy", we could lose metrics for previous stages of the subquery.

Reply:
+1, some maps in LiveStageMetrics will lose

Reply:
After discussion with @cloud-fan , we update the sql metric in` LiveExecutionData.metirc` and here no need to copy. Thanks.